File: 프로그래머스/unrated/181873. 특정한 문자를 대문자로 바꾸기/특정한 문자를 대문자로 바꾸기.java
class Solution {
    public String solution(String my_string, String alp) {
       String answer ="";
        for(int i=0; i<my_string.length(); i++){
            String str = my_string.charAt(i)+"";
            if(str.equals(alp)){
                str = str.toUpperCase();
            }
              answer += str;          
        }
        return answer;
    }
}


//첫풀이
//class Solution {
//    public String solution(String my_string, String alp) {
       
//        return my_string.replace(alp, alp.toUpperCase());
//    }
//}

File: 프로그래머스/unrated/181873. 특정한 문자를 대문자로 바꾸기/특정한 문자를 대문자로 바꾸기.js
function solution(my_string, alp) {
    var answer ='';
    var str = my_string.split("");
    
    for(var i=0; i<str.length; i++){
        if(str[i] === alp){
            answer += alp.toUpperCase();
        }else{
            answer += str[i];
        }
    }
    return answer;
}